doc/src/modules/matrices/random.rst

```diff
@@ -0,0 +1,371 @@
+      :py:func:`jordan_normal <sympy.matrices.random.jordan_normal>`
+
+    - :py:func:`nilpotent <sympy.matrices.random.nilpotent>` with
+      :py:func:`jordan_normal <sympy.matrices.random.jordan_normal>` (with zero eigenvalues)
+
+    *isometry conjugate matrices* that arise as a product $\mathbf{OAO}^{-1} = \mathbf{OAO}^{T}$
+    of specific matrices $A$
+    where $\mathbf{A}$ sets the (complex) eigenvalue spectrum.
+
+    - :py:func:`orthogonal <sympy.matrices.random.orthogonal>`
+      can be a conjugate of
+      :py:func:`isometry_normal <sympy.matrices.random.isometry_normal>`
+      by
+      :py:func:`orthogonal <sympy.matrices.random.orthogonal>`
+
+    - same holds for :py:func:`unitary <sympy.matrices.random.unitary>`
+      which can be a conjugate of
+      :py:func:`isometry_normal <sympy.matrices.random.isometry_normal>`
+      by
+      :py:func:`unitary <sympy.matrices.random.unitary>`
+
+    - :py:func:`normal <sympy.matrices.random.normal>`
+      is a conjugate of
+      :py:func:`diagonal_normal <sympy.matrices.random.diagonal_normal>`
+      by either
+      :py:func:`orthogonal <sympy.matrices.random.orthogonal>`
+      or
+      :py:func:`unitary <sympy.matrices.random.unitary>`
+
+    finally *symmetric matrices* given as a as product $\mathbf{MM}^T$
+    resp. $\mathbf{MM}^H$ of $\mathbf{M}$, such as
+
+    - :py:func:`symmetric <sympy.matrices.random.symmetric>` $\mathbf{MM}^T$ with
+      :py:func:`invertible <sympy.matrices.random.invertible>` $\mathbf{M}$
+
+    - :py:func:`hermite <sympy.matrices.random.hermite>` $\mathbf{MM}^H$ with
+      :py:func:`invertible <sympy.matrices.random.orthogonal>` $\mathbf{M}$
+
+
+In addition to the type of matrix, also the type of entries (as a commutative ring with one)
+to be able to define values (**scalars**) can be specified,
+from which the value entries (**scalar**) of the basic matrices are randomly generated.
+
+Because the complexity and amount of entries in the generated compound matrices
+in addition to the **scalars**, also by the number of base matrices multiplied for generation
+This can be set using the argument **length**.
+
+Normal forms as well as conjugate types have the arg **spec** to provide a spectrum of eigenvalues.
+
+The product of *elementary* matrices
+
+    >>> from sympy.matrices.random import elementary
+
+    >>> A = elementary(3, (0, 2), -2)
+    >>> A
+    Matrix([
+    [1, 0, -2],
+    [0, 1,  0],
+    [0, 0,  1]])
+
+    >>> B = elementary(3, (1, 2), 1)
+    >>> B
+    Matrix([
+    [1, 0, 0],
+    [0, 1, 1],
+    [0, 0, 1]])
+
+    >>> C = elementary(3, (1, 0), -1)
+    >>> C
+    Matrix([
+    [ 1, 0, 0],
+    [-1, 1, 0],
+    [ 0, 0, 1]])
+
+gives an *invertible* matrix $\mathbf{M}$
+
+    >>> M = A*B*C
+    >>> M
+    Matrix([
+    [ 1, 0, -2],
+    [-1, 1,  1],
+    [ 0, 0,  1]])
+
+    >>> M.inv()
+    Matrix([
+    [1, 0, 2],
+    [1, 1, 1],
+    [0, 0, 1]])
+
+
+Seeding Random Number Generators
+--------------------------------
```

Comment:
In the time since this PR was created a new `sympy/core/random` module has been added so that there is centralised control of seeding for random function in SymPy:
https://github.com/sympy/sympy/blob/4489b4c7c51a65966b1a95f2af6f7e0c8c461794/sympy/core/random.py#L1-L10

Reply:
This dosen't work properly. See #22592.